forest/cli/src/cli/snapshot_cmd.rs

```diff
@@ -323,6 +348,125 @@ fn clean(config: &Config, snapshot_dir: &Option<PathBuf>, force: bool) -> anyhow
     Ok(())
 }
 
+async fn validate(
+    config: &Config,
+    validate_height: &i64,
+    snapshot: &PathBuf,
+) -> anyhow::Result<()> {
+    let confirm = Confirm::with_theme(&ColorfulTheme::default())
+        .with_prompt(format!(
+            "This will result in using approximately {} MB of data. Proceed?",
+            std::fs::metadata(snapshot).unwrap().len() / (1024 * 1024)
+        ))
+        .default(false)
+        .interact()
+        .unwrap_or_default();
+
+    if confirm {
+        // clear temp db if exists
+        let tmp_db_path = PathBuf::from("/tmp/validate_snapshot");
```

Comment:
You can use the `tempfile` crate to get a temporary directory which will be deleted when the handle is dropped.

Reply:
ah nice, added